chain/chain/src/chain.rs

```diff
@@ -5099,7 +5102,17 @@ impl<'a> ChainUpdate<'a> {
 pub fn do_apply_chunks(
     work: Vec<Box<dyn FnOnce() -> Result<ApplyChunkResult, Error> + Send>>,
 ) -> Vec<Result<ApplyChunkResult, Error>> {
-    work.into_par_iter().map(|task| task()).collect::<Vec<_>>()
+    let parent_span = tracing::debug_span!(target: "chain", "Apply chunks").entered();
+    work.into_par_iter()
+        .map(|task| {
+            // As chunks can be processed in parallel, make sure they are all tracked as children of
+            // a single span.
+            let _span =
+                tracing::debug_span!(target: "chain", parent: &parent_span, "Chunk").entered();
```

Comment:
```suggestion
                tracing::debug_span!(target: "chain", parent: &parent_span, "Apply chunk").entered();
```

Another thing that'd be great to see documented in the style guide: do we name our spans as a single noun describing what is being processed or do we go more verbose and actually add a verb in there as well. We should apply either one or the other consistently.

---

Another alternative is to use hierarchies like this:

```
Blocks -> Block -> Preprocessing
                -> Postprocessing
                -> ...
Chunks -> Preprocessing
       -> Postprocessing
       -> Chunk
       -> ...
```

and while it is much more compact and readable in certain scenarios, it also requires having access to contextual information to interpret fully (“preprocessing” on its own doesn't really tell you much about what is being preprocessed.) Not a huge deal in prometheus or certain logging set-ups, but might be subpar in some other scenarios.